service/lib/agama/dbus/storage/volume_conversion/from_dbus.rb

```diff
@@ -0,0 +1,147 @@
+                next unless converter
+
+                send(converter, target, dbus_value)
+              end
+            end
+          end
+
+        private
+
+          # @return [Hash]
+          attr_reader :dbus_volume
+
+          # @return [Agama::Config]
+          attr_reader :config
+
+          # D-Bus attributes and their converters.
+          CONVERSIONS = {
+            "MountPath"    => :mount_path_conversion,
+            "MountOptions" => :mount_options_conversion,
+            "TargetDevice" => :target_device_conversion,
+            "TargetVG"     => :target_vg_conversion,
+            "FsType"       => :fs_type_conversion,
+            "MinSize"      => :min_size_conversion,
+            "MaxSize"      => :max_size_conversion,
+            "AutoSize"     => :auto_size_conversion,
+            "Snapshots"    => :snapshots_conversion
+          }.freeze
+          private_constant :CONVERSIONS
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [String]
+          def mount_path_conversion(target, value)
+            target.mount_path = value
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [Array<String>]
+          def mount_options_conversion(target, value)
+            target.mount_options = value
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [String]
+          def target_device_conversion(target, value)
+            target.device = value
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [String]
+          def target_vg_conversion(target, value)
+            target.separate_vg_name = value
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [String]
+          def fs_type_conversion(target, value)
+            fs_type = target.outline.filesystems.find do |type|
+              type.to_human_string.downcase == value.downcase
+            end
+
+            return unless fs_type
+
+            target.fs_type = fs_type
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [Integer]
+          def min_size_conversion(target, value)
+            target.min_size = Y2Storage::DiskSize.new(value)
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [Integer]
+          def max_size_conversion(target, value)
+            target.max_size = Y2Storage::DiskSize.new(value)
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [Boolean]
+          def auto_size_conversion(target, value)
+            return unless target.auto_size_supported?
+
+            target.auto_size = value
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [Booelan]
+          def snapshots_conversion(target, value)
+            return unless target.outline.snapshots_configurable?
```

Comment:
I would say that for both case where we skip conversion it should be logged what happen.